ruma-api/src/lib.rs

```diff
@@ -267,6 +267,26 @@ pub struct Metadata {
     pub requires_authentication: bool,
 }
 
+/// Generates a `match` statement for the given expression that returns early.
+#[macro_export]
+macro_rules! try_deserialize {
+    ($kind:ident, $call:expr $(,)*) => {
```

Comment:
To not accept arbitrarily many commas at the end, you can write

```suggestion
    ($kind:ident, $call:expr $(,)?) => {
```

(since Rust 1.32)

Reply:
Good to know thanks!